### PR TITLE
`graph_mask_t` and separating raft includes for `host_span` and `device_span` 

### LINKED ISSUE #2322: [FEA] Prototype vertex and edge masking approach
#2218 defined the context of the problem and some initial designs.

The goal in 22.08 is to prototype something so that we can build a capability to deliver in 22.10.

### PR DESCRIPTION
**Pull request by cjnolet**
This depends on https://github.com/rapidsai/raft/pull/810. 

Closes #2322

### CODE REVIEW

**Comment by cjnolet** [comment]
rerun tests

File: cpp/include/cugraph/graph_mask.hpp [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+}
+
+/**
+ * Uses bit-shifting to perform a fast mod operation. This
+ * is used to compute the index of a specific bit
+ * @tparam mask_t
+ * @tparam T
+ */
+template <typename mask_t, typename T>
+__host__ __device__ int bit_mod(T numer)
```
Comment (seunghwak) [inline cpp/include/cugraph/graph_mask.hpp:54]: I assume `numer` here is a typo of `number`.

File: cpp/cmake/thirdparty/get_raft.cmake [hunk 2]
```diff
@@ -55,8 +55,8 @@ endfunction()
 # To use a different RAFT locally, set the CMake variable
 # CPM_raft_SOURCE=/path/to/local/raft
 find_and_configure_raft(VERSION    ${CUGRAPH_MIN_VERSION_raft}
-                        FORK       rapidsai
-                        PINNED_TAG branch-${CUGRAPH_BRANCH_VERSION_raft}
+                        FORK       cjnolet
+                        PINNED_TAG imp-2210-host_device_mdspan
```
Comment (seunghwak) [inline cpp/cmake/thirdparty/get_raft.cmake:59]: Don't forget to revert this.

**Review by seunghwak — APPROVED** [review]
Looks good to me (but I assume this should be merged after the raft side PR gets merged and the change in `get_raft.cmake` gets reverted.

File: cpp/include/cugraph/graph_mask.hpp (hunk 1 above)
Comment (cjnolet) [reply cpp/include/cugraph/graph_mask.hpp:54]: It's actually the `numerator` of the division for the mod but i can change it to `number` if that's what you prefer

Comment (seunghwak) [reply cpp/include/cugraph/graph_mask.hpp:54]: I see, then, this is not a big issue, but I slightly prefer either `number` or `numerator` than `numer`. It is little difficult to infer `numerator` from `numer`.

**Comment by cjnolet** [comment]
rerun tests

**Comment by cjnolet** [comment]
rerun tests

**Review by ChuckHastings — APPROVED** [review]

**Review by rlratzel — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

**Comment by rlratzel** [comment]
@gpucibot merge

**Pull request merged**